registry/registry.go

```diff
@@ -332,9 +333,13 @@ func (registry *Registry) ListenAndServe() error {
 	}
 }
 
-// Shutdown gracefully shuts down the registry's HTTP server.
+// Shutdown gracefully shuts down the registry's HTTP server and application object.
 func (registry *Registry) Shutdown(ctx context.Context) error {
-	return registry.server.Shutdown(ctx)
+	err := registry.server.Shutdown(ctx)
+	if appErr := registry.app.Shutdown(); appErr != nil {
```

Comment:
We call registry.app.Shutdown and that function gets the registry from the app, checks the type, and closes it.

Could we just call registry.Close here and not need the app shutdown method at all?

Reply:
Yes it is confusing, perhaps the app struct entry should be namespace and not registry.

But you're absolutely right, the implementation is correct.